### PR TITLE
Validate Alertmanager config

### PR DESCRIPTION
**Pull request by allangood**
Install the amtool and validate alertmanager config file prior changes.

#### Pull Request (PR) description
Validate Alertmanager config file prior changes

#### This Pull Request (PR) fixes the following issues
Alertmanager has a validation tool but it's not installed nor used by this modules until now.

### CODE REVIEW

**Comment by allangood** [comment]
Fixed require parameter and cleaned all lint warnings.

File: manifests/alertmanager.pp [hunk 1]
```diff
@@ -184,13 +190,14 @@
...
-    notify  => $notify_service,
-    require => File[$config_dir],
+    ensure       => present,
+    owner        => $user,
+    group        => $group,
+    mode         => $config_mode,
+    content      => template('prometheus/alertmanager.yaml.erb'),
+    notify       => $notify_service,
+    require      => File["${bin_dir}/amtool", $config_dir],
+    validate_cmd => "${bin_dir}/amtool check-config --alertmanager.url='' %",
```
Comment (bastelfreak) [inline manifests/alertmanager.pp]: do you know if this works for different versions of the alertmanager? Can you provide an acceptance test for this?

Comment (allangood) [reply manifests/alertmanager.pp]: Doing a quick check on Alertmanager Github page, the amtool command was implemented in version 0.10.0. I could implement a "versioncmp" to make sure to do the validation check if the version is equal or grater than 0.10.0. This works for you?

Comment (bastelfreak) [reply manifests/alertmanager.pp]: sounds good!

**Review by allangood — COMMENTED** [review]
I just realized that I've introduced an old behaviour in the "notify" parameter. Let me correct it.

**Comment by allangood** [comment]
Nothing left from my part.

**Comment by bastelfreak** [comment]
@allangood can you add an acceptance test for this?

**Comment by allangood** [comment]
Never did it before, I will do my best using "prometheus_server_spec.rb" as template.

**Comment by bastelfreak** [comment]
Ping @allangood :)

**Comment by allangood** [comment]
Sorry about this long silence... I started to work with something else and completely forgot to commit this code.
Sorry again.

**Comment by allangood** [comment]
Well, looks like I need some guidance with the acceptance test! Someone could give me a hand?

File: spec/acceptance/alertmanager_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+require 'spec_helper_acceptance'
+
+describe 'prometheus alertmanager' do
+  it 'alertmanager works idempotently with no errors' do
+    pp = "class { 'prometheus::alertmanager': }"
```
Comment (bastelfreak) [inline spec/acceptance/alertmanager_spec.rb]: the previous style was correct, you can undo the last commit.

**Comment by bastelfreak** [comment]
Hi. The test itself works. Puppet creates the service `alertmanager`, it's also enabled. It fails to start during the first puppet run and also during the second. that's why there isn't an open TCP port. Did you test this locally? I assume the generated config isn't valid or something is missing.

**Comment by bastelfreak** [comment]
Thanks for all the work @allangood

**Comment by allangood** [comment]
Wow! It's working now! I had to include some configuration to Alertmanager, the default "include prometheus::alermanager" is not enough.

Sorry for the amount of commits. :)

**Pull request merged**